src/com/xilinx/rapidwright/router/UltraScaleClockRouting.java

```diff
@@ -125,6 +126,117 @@ public static RouteNode routeToCentroid(Net clk, RouteNode clkRoutingLine, Clock
 		}
 		return null;		
 	}
+	
+	/**
+	 * Routes a clock from a routing track to a given transition point called the centroid
+	 * @param clk The clock net to be routed
+	 * @param startingRouteNode The starting routing track
+	 * @param centroid The given centroid node
+	 * @return
+	 */
+	public static RouteNode routeToGivenCentroidNode(Net clk, RouteNode startingRouteNode, Node centroid) {
```

Comment:
Perhaps refactor this to `routeToCentroidNode()`

Reply:
Refactored.